ledger/db/src/tx_out_store.rs

```diff
@@ -1127,23 +1123,24 @@ pub mod tx_out_store_tests {
                                |                                  |
                             H(0,3)                       ______ H(4,7)_____
                                                          |                |
-                                                   ___H(4,5)__          H(6,7) = H(Nil)
+                                                   ___H(4,5)__          H(6,7) = H(Nil) Omitted
                                                    |         |
                                                  H(4,4)   H(5,5)
                                                    |        |
                    0       1       2       3       4       [5]       Nil       Nil
 
             */
 
-            assert_eq!(7, proof.elements.len());
+            assert_eq!(6, proof.elements.len());
```

Comment:
The point of omitting Nil hashes is to reduce the size of the Merkle proof. Here, the proof size is reduced by about 14%.